pennylane/devices/default_qubit.py

```diff
@@ -468,8 +468,8 @@ def expval(self, observable, shot_range=None, bin_size=None):
         Returns:
             float: returns the expectation value of the observable
         """
-        # intercept Hamiltonians here; in future, we want a logic that handles
-        # general observables that do not define eigenvalues
+        # intercept other Hamiltonians
+        # TODO: find a better way, in which devices do not have to reproduce this logic.
```

Comment:
cosmetic change to leave a better message for future developers

Reply:
```suggestion
        # TODO: Ideally, this logic should not live in the Device, but be moved
        # to a component that can be re-used by devices as needed.
```